catalyst/utils/components.py

```diff
@@ -65,6 +65,17 @@ def process_components(
 
     model: Model = maybe_recursive_call(model, "to", device=device)
 
+    if optimizer is not None:
+        if isinstance(optimizer, dict):
+            optimizer = {
+                k: type(opt)(model[k].parameters(), **opt.defaults)
+                for k, opt in optimizer.items()
+            }
+        else:
+            optimizer = type(optimizer)(
+                model.parameters(), **optimizer.defaults
+            )
```

Comment:
really tricky solution :)
I understand why, but... are you sure, that opt.defaults have all required parameters?

Reply:
I think it will work in most cases.
I've checked optimizers in pytorch and saw that all arguments wraps in defaults dict and used in initialization of super class [Optimizer](https://pytorch.org/docs/stable/optim.html#torch.optim.Optimizer)

But this approach is bad because current implementation don't know anything about user defined learning rates for layers and we need to somehow handle this case.